hail_scripts/reference_data/combine.py

```diff
@@ -49,14 +40,17 @@ def get_select_fields(selects, base_ht):
     elif isinstance(selects, dict):
         for key, val in selects.items():
             # Grab the field and continually select it from the hail table.
-            ht = base_ht
+            expression = base_ht
             for attr in val.split('.'):
                 # Select from multi-allelic list.
                 if attr.endswith('#'):
-                    ht = ht[attr[:-1]][base_ht.a_index - 1]
+                    expression = expression[attr[:-1]][base_ht.a_index - 1]
                 else:
-                    ht = ht[attr]
-            select_fields[key] = ht
+                    expression = expression[attr]
+            # Parse float64s into float32s to save space!
+            if expression.dtype == hl.tfloat64:
```

Comment:
Should this be bumped under the for loop? Though I guess we will always have values so expression=base_ht is not going to run right before this line in any case?

Reply:
I think the idea is we only need to convert to `hl.tfloat32` a single time rather than for each of the nested expressions.

Reply:
Ah that makes sense!